pre_commit/main.py

```diff
@@ -136,6 +136,15 @@ def _add_run_options(parser: argparse.ArgumentParser) -> None:
             'file from the index, flag=0).'
         ),
     )
+    parser.add_argument(
+        '--is-squash-merge',
+        action='store_const',
+        const='1',
```

Comment:
I think this should probably be `'--squash-merge', default='', action='store_const', const='1'`

though it might make sense to make this a boolean?  and then we'd parse the incoming value from git in the `hook-impl` command (maybe `== '1'`? does git specify what values it'll send here?)

Reply:
Git sends `'0'` and  `'1'` for `post-merge` hooks. This is true for `post-checkout` as well, with the `--checkout-type` argument. The existing `checkout-type` flag does not supply a default, though plausibly it should default to `'1'`. I will use your suggestion, except with default `'0'`

Reply:
ah, I ended up changing this to be more like the `--checkout-type` argument -- that way you can simulate both `=1` and `=0` -- as it was before I fixed it it was always setting the `PRE_COMMIT_IS_...` variable which was not quite right (since `'0'` is truthy)